pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -183,12 +183,20 @@ func getBoolOption(name string, defval bool, options map[string]interface{}) boo
 
 // Manifests returns r, which implements distribution.ManifestService.
 func (r *repository) Manifests(ctx context.Context, options ...distribution.ManifestServiceOption) (distribution.ManifestService, error) {
+	if err := r.checkPendingErrors(ctx); err != nil {
```

Comment:
Let's propagate the evaluation to leaves. We'll want to decide what to do with them in individual methods like `blobStore.Get()` (don't authorize against namespace but rather see if mounted blob is present there).